plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -115,6 +117,9 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
         if (fromType instanceof VarcharType fromVarcharType && toType instanceof DateType toDateType) {
             return Optional.of(new VarcharToDateCoercer(fromVarcharType, toDateType));
         }
+        if (fromType instanceof BooleanType && toType instanceof VarcharType toVarcharType) {
+            return Optional.of(new BooleanToVarcharCoercer(toVarcharType));
```

Comment:
Should we return Optional.empty when the varchar length < 4? Sorry, I'm not familiar with this class. Asking this question as other conditions look returning the empty object instead of throwing an exception.

Reply:
In such situations, the code base fails (instead of silently returning empty) with a `TrinoException` 

```
throw new TrinoException(INVALID_ARGUMENTS, format("Varchar representation of %s exceeds %s bounds", value, toType));
```

Reply:
My question was why an exception is better than empty. Other conditions seem returning empty like:
```java
        if (fromType instanceof CharType fromCharType && toType instanceof CharType toCharType) {
            if (narrowerThan(toCharType, fromCharType)) {
                return Optional.of(new CharCoercer(fromCharType, toCharType));
            }
            return Optional.empty();
        }
```

Reply:
I'm not familiar enough with the functionality, but I'm assuming we should fail early instead of doing truncation like Hive does:

```
0: jdbc:hive2://localhost:10000/default> select * from testboolean;
+----------------+
| testboolean.b  |
+----------------+
| true           |
| fals           |
| NULL           |
```


cc @Praveen2112 @homar

Reply:
Actually I just stumbled on `timestamp` to `varchar` coercion where Trino does truncate as well 

https://github.com/trinodb/trino/blob/e77b84893658dd422df9677bb3b3e5b9ccad7cd9/plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestTimestampCoercer.java#L74-L93

I'm a bit confused now - I know the point of discussion is a corner case, but I'd like to have a clear understanding on this topic.

cc @Praveen2112

Reply:
@findepi

Reply:
I think we can throw an explicit exception in case of lower bounded varchar.